ako-infra/ingestion/avi_infra_controller.go

```diff
@@ -123,42 +192,45 @@ func (a *AviControllerInfra) DeriveCloudNameAndSEGroupTmpl(tz string) (error, st
 			continue
 		}
 		utils.AviLog.Infof("Found NSX-T cloud: %s match Transport Zone: %s", *cloud.Name, tz)
-		if cloud.SeGroupTemplateRef != nil && *cloud.SeGroupTemplateRef != "" {
-			tokenized := strings.Split(*cloud.SeGroupTemplateRef, "/api/serviceenginegroup/")
-			if len(tokenized) == 2 {
-				return nil, *cloud.Name, tokenized[1]
-			}
-		}
-
-		// fetch Default-SEGroup uuid
-		uri = "/api/serviceenginegroup/?include_name&cloud_ref.name=" + *cloud.Name + "&name=Default-Group"
-		result, err := lib.AviGetCollectionRaw(a.AviRestClient, uri)
-		if err != nil {
-			utils.AviLog.Errorf("Get uri %v returned err %v", uri, err)
-			return err, *cloud.Name, ""
+		matchCloud = cloud
+		break
+	}
+	if matchCloud == nil {
+		return errors.New("cloud not found matching transport zone " + tz), "", ""
+	}
+	if matchCloud.SeGroupTemplateRef != nil && *matchCloud.SeGroupTemplateRef != "" {
+		tokenized := strings.Split(*matchCloud.SeGroupTemplateRef, "/api/serviceenginegroup/")
+		if len(tokenized) == 2 {
+			return nil, *matchCloud.Name, tokenized[1]
```

Comment:
Do we need to bring this logic out of the loop? don't think we need matchCloud flag either.

Reply:
For code readability, I moved all the other things out of the loop. The logic is same, just the loop gets to exit or break early.